app/src/org/odk/collect/android/logic/ArchivedFormRemoteRestore.java

```diff
@@ -0,0 +1,131 @@
+package org.odk.collect.android.logic;
+
+import android.widget.Toast;
+
+import org.commcare.android.database.user.models.User;
+import org.commcare.android.framework.CommCareActivity;
+import org.commcare.android.tasks.DataPullTask;
+import org.commcare.android.tasks.FormRecordCleanupTask;
+import org.commcare.android.tasks.templates.CommCareTaskConnector;
+import org.commcare.android.util.SessionUnavailableException;
+import org.commcare.dalvik.activities.FormRecordListActivity;
+import org.commcare.dalvik.application.CommCareApplication;
+import org.commcare.util.CommCarePlatform;
+import org.javarosa.core.services.locale.Localization;
+
+/**
+ * Only used for developer debugging.
+ *
+ * Load saved form instances manually from a xml payload.
+ *
+ * @author Phillip Mates (pmates@dimagi.com).
+ */
+public class ArchivedFormRemoteRestore<A extends CommCareActivity> {
```

Comment:
@amstone326 removed the generics, should be good to go.
Realized that this should be occurring in a background task (Kriti was complaining about recent slowness). I'll get that going, but it should go into a separate PR